docs/docs/topics/mutual-auth.md

```diff
@@ -0,0 +1,390 @@
+
+```mermaid
+flowchart LR
+  A(End User)
+  B(Pomerium)
+  C(Sidecar)
+  D(Service)
+  E[/Hacker/]
+  subgraph lan[Internal Network]
+    style lan stroke-dasharray: 5 5
+    subgraph spacing[ ]
+      style spacing stroke-width:0px,fill:#ffffde;
+    end
+    subgraph docker1 [Docker Network Frontend]
+      style docker1 stroke-dasharray:5 5;
+      B
+    end
+    subgraph docker2 [Docker Network Backend]
+      C
+      D
+    end
+  end
+  A==oB
+  B<==>C
+  C---D
+  E---xdocker2
+```
+
+## Putting It All Together
+
+Each of the techniques described above provides an additional form of security. Which you choose to apply to your infrastructure depends on your needs, end users, and what your services and/or infrastructure can support.
+
+Security practices can often seem like a scale with best practices at one end and ease of use at the other. Pomerium is designed to make it reasonably easy to enable the best security model that fits each service. For example, it's usually not reasonable to expect a service with many unique end users to require client certificates for downstream mTLS, but communication between Pomerium and the upstream application or service mesh can use it without any additional overhead for the client.
+
+At this point, the security model becomes difficult to accurately describe with a simple graph... but we'll try anyway.
+
+```mermaid
+flowchart LR
+  A(End User)
+  B(Pomerium)
+  C(Identity Provider)
+  D(User App)
+  E(Admin Services)
+  F(Sidecar)
+  G(Admin)
+  H(API)
+  subgraph lan[Internal Network]
+    style lan stroke-dasharray: 5 5
+    direction LR
+    B
+    D
+    E
+    F
+    subgraph Sidecar [Secure VLAN];
+      H
+      F---H
+    end
+  end
+  A ==TLS==o B
+  B<==mTLS==>D
+  B-.JWT.-D
+  B<==mTLS==>E
+  B<==mTLS===>F
+  G<==mTLS==>B
+  C-.-B
+```
+
+- In this example, End users authenticate with Pomerium to access the user app:
+  - Their connection to Pomerium is authenticated after signing in to their identity provider, and encrypted with TLS.
+  - The communication between Pomerium and the user app is mutually authenticated with mTLS, and the user's JWT is passed from Pomerium to the app to confirm the user identity to the service. The service validates the JWT against Pomerium's signing Key.
+- The admin also authenticates with Pomerium to access their services:
+  - The connection is authenticated and encrypted the same as the user's, but the route to the admin app also requires a client certificate for mutual authentication at the protocol layer.
+  - The connections from the admin to the admin services are now mutually authenticated both upstream and downstream from Pomerium.
+- The API server is required by both the user and admin apps. Rather than build authentication into the API service, it is accessible only through a sidecar. The sidecar only accepts mTLS-authenticated connections from the proxy, so both apps connect to the API through Pomerium.
+
+## Legend
```

Comment:
Could we move this to the top of the doc?

Reply:
Discussed in Slack.